pkg/auth/context/context.go

```diff
@@ -0,0 +1,21 @@
+package auth
+
+import "context"
+
+// SAAuthenticatedContextKey is the context key for the SAAuthenticated flag.
+var SAAuthenticatedContextKey = struct{}{}
+
+// IsSAAuthenticated returns true if the SAAuthenticated flag is set in the context.
+func IsSAAuthenticated(ctx context.Context) bool {
+	authed, ok := ctx.Value(SAAuthenticatedContextKey).(bool)
+	if !ok {
+		return false
+	}
+
+	return authed
```

Comment:
Since we are defaulting to false we the assertion check can be skipped (not avoided, or it will panic otherwise):

```suggestion
	authed, _ := ctx.Value(SAAuthenticatedContextKey).(bool)
	return authed
```

Reply:
Ok, seems reasonable.